include/drivers/dai.h

```diff
@@ -0,0 +1,717 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * TODO:
+ *
+ * 1) Extend to support DMIC, HDA and ALH types from SOF (check the IPC headers
+ *    from AMD, NXP, Mediatek too for any quirks/types that need to be added).
+ * 2) Support early/late BCLKs for SSPs.
+ * 3) Dont break backwards compat for older Zephyr code using i2s.h to transition
+ *    to this DAI API.
+ * 4) Fix comment to remove I2S refs.
+ * 5) Add any aliases for I2S/SSP format modes to ensure legacy Ze[hyr and SOF
+ *    compat.
+ * 6) make the mem_slab mode use by i2s.h as optional ops.
+ * 7) Add any ops/data to help with integration with DMA.
+ */
+
+/**
+ * @file
+ * @brief Public APIs for the DAI (Digital Audio Interface) bus drivers.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_DAI_H_
+#define ZEPHYR_INCLUDE_DRIVERS_DAI_H_
+
+/**
+ * @defgroup dai_interface DAI Interface
+ * @ingroup io_interfaces
+ * @brief DAI (Inter-IC Sound) Interface
+ *
+ * The DAI API provides support for the standard DAI interface standard as well
+ * as common non-standard extensions such as PCM Short/Long Frame Sync,
+ * Left/Right Justified Data Format.
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * The following #defines are used to configure the DAI controller.
+ *
+ * TODO converge with IPC headers.
+ */
+
+
+typedef uint8_t dai_fmt_t;
+
+/** Data Format bit field position. */
+#define DAI_FMT_DATA_FORMAT_SHIFT           0
+/** Data Format bit field mask. */
+#define DAI_FMT_DATA_FORMAT_MASK            (0x7 << DAI_FMT_DATA_FORMAT_SHIFT)
+
+
+/** @brief Standard DAI Data Format.
+ *
+ * Serial data is transmitted in two's complement with the MSB first. Both
+ * Word Select (WS) and Serial Data (SD) signals are sampled on the rising edge
+ * of the clock signal (SCK). The MSB is always sent one clock period after the
+ * WS changes. Left channel data are sent first indicated by WS = 0, followed
+ * by right channel data indicated by WS = 1.
```

Comment:
We can put this into another ABI file, but these are very useful for developers as this HW interface has to be aligned between the SOC and the Codec/BT/MODEM. Some of the devices (like BT and MODEMs) run closed FW where it's really only possible to figure out the interface with a scope (and compare and align with the SOC here).

Reply:
Since there are so many integration methods for DAIs, those kind of details may apply to a very small subset while the others are programmed elsewhere, outside the DSP. Therefore not sure if this piece of the documentation should be included in this very common definition.